server/http.go

```diff
@@ -219,18 +219,79 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if webhook.Event != zoom.EventTypeMeetingEnded {
-		w.WriteHeader(http.StatusNotImplemented)
+	if webhook.Event == zoom.EventTypeMeetingParticipantJoined {
+		var meetingWebhook zoom.MeetingParticipantsJoinedWebhook
+		if err = json.Unmarshal(b, &meetingWebhook); err != nil {
+			p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		p.handleParticipantJoinedWebhook(w, r, &meetingWebhook)
+		return
+	}
+	if webhook.Event == zoom.EventTypeMeetingEnded {
+		var meetingWebhook zoom.MeetingWebhook
+		if err = json.Unmarshal(b, &meetingWebhook); err != nil {
+			p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		p.handleMeetingEnded(w, r, &meetingWebhook)
 		return
 	}
+	http.Error(w, "Not implemented", http.StatusNotImplemented)
+}
 
-	var meetingWebhook zoom.MeetingWebhook
-	if err = json.Unmarshal(b, &meetingWebhook); err != nil {
-		p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
-		http.Error(w, err.Error(), http.StatusBadRequest)
+// handle webhook participant join before host in the meeting
+func (p *Plugin) handleParticipantJoinedWebhook(
+	w http.ResponseWriter, r *http.Request, webhook *zoom.MeetingParticipantsJoinedWebhook,
+) {
+	meetingID := webhook.Payload.Object.ID
+	postID, appErr := p.fetchMeetingPostID(meetingID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
 		return
 	}
-	p.handleMeetingEnded(w, r, &meetingWebhook)
+	post, appErr := p.API.GetPost(postID)
+	if appErr != nil {
+		p.API.LogWarn("Could not get meeting post by id", "err", appErr)
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+	meetingCreator := post.UserId
+	participant := webhook.Payload.Object.Participant
+	isHostJoined, isBool := post.Props["meeting_host_joined"].(bool)
+	if !isBool {
+		isHostJoined = false
+	}
+	waitingCnt, isNum := post.Props["meeting_waiting_count"].(float64)
+	if !isNum {
+		waitingCnt = 0
+	}
+	// if host has joined, then no need to proceed further
+	if isHostJoined {
+		return
+	}
+	// check whether participant is host
+	if participant.RegistrantID == webhook.Payload.Object.HostID {
+		isHostJoined = true
+	} else {
+		waitingCnt++
+		p.API.SendEphemeralPost(meetingCreator, &model.Post{
+			UserId:    p.botUserID,
+			ChannelId: post.ChannelId,
+			Message: fmt.Sprintf(
+				"User %s has joined the meeting before you", participant.UserName,
+			),
+		})
+	}
+	post.Props["meeting_host_joined"] = isHostJoined
+	post.Props["meeting_waiting_count"] = waitingCnt
+	_, updatePostErr := p.API.UpdatePost(post)
+	_, err := w.Write([]byte(post.ToJson()))
```

Comment:
I wonder if there is any need to write the post to the zoom. Maybe just an OK is enough?

Reply:
Yeah, I could fix that =))